execution_engine/src/core/runtime/mint_internal.rs

```diff
@@ -189,4 +189,31 @@ where
     R: StateReader<Key, StoredValue>,
     R::Error: Into<execution::Error>,
 {
+    /// Burns native tokens.
+    fn burn(&mut self, purse: URef, amount: U512) -> Result<(), Error> {
+        let purse_key = Key::URef(purse);
+        self.context
+            .validate_writeable(&purse_key)
+            .map_err(|_| Error::InvalidAccessRights)?;
+        self.context
+            .validate_key(&purse_key)
+            .map_err(|_| Error::InvalidURef)?;
```

Comment:
I think this should be a ForgedReference error...

https://github.com/casper-network/casper-node/blob/feat-2.0/types/src/system/mint/error.rs#L161